### PR TITLE
Issue #159: Implement `schedule edit` command

### LINKED ISSUE #159: Add `schedule edit` command
Note: Only the interview date and/or time is allowed to be changed. 
Format:
`schedule edit SCHEDULE_INDEX NEW_DATE_TIME`
Example:
`schedule edit 2 13-05-2022 10:00`

### PR DESCRIPTION
**Pull request by lzan98**
Hey y'all! After contemplating if I should go with `schedule edit INDEX DATE_TIME` or  `schedule edit INDEX at/DATE_TIME`, I decided to go with the one with the prefix because the original one is prone to bugs such as `IndexOutOfBoundsException` or the code will look hard-coded/amateurish. If y'all want to stick to the old one lmk, I can change it back!

Closes #159

### CODE REVIEW

File: src/main/java/seedu/address/model/InterviewSchedule.java [hunk 1]
```diff
@@ -81,12 +81,11 @@ public void addInterview(Interview interview) {
         interviews.add(interview);
     }
 
-    // Edit function not implemented yet, code commented out to reduce test coverage.
-    /* public void setInterview(Interview target, Interview editedInterview) {
+    public void setInterview(Interview target, Interview editedInterview) {
```
Comment (charmainehly) [inline src/main/java/seedu/address/model/InterviewSchedule.java:84]: Thanks for the hard work on the `Schedule` command 🥲 !!!!!!

Just to check, does the `schedule edit` command disallow interviews to be set in the past - sorry not sure if i missed out the implementation for this.

**Review by charmainehly — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/InterviewSchedule.java (hunk 1 above)
Comment (lzan98) [reply src/main/java/seedu/address/model/InterviewSchedule.java:84]: Hey char! No worries, the `parseDateTime` function in `EditScheduleCommandParser| takes care of this issue, just like in `AddScheduleCommandParser`! `parseDateTime` checks for both formatting and whether the input date time is in the past.

Comment (charmainehly) [reply src/main/java/seedu/address/model/InterviewSchedule.java:84]: Ah I see!! :) Okay thanks! 👍

**Review by charmainehly — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/schedule/EditScheduleCommand.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return other == this // short circuit if same object
+                || (other instanceof EditScheduleCommand // instanceof handles nulls
+                && index.equals(((EditScheduleCommand) other).index)); // state check
+    }
+
+    /**
+     * Creates and returns a {@code Candidate} with the details of {@code candidateToEdit}
+     * edited with {@code editCandidateDescriptor}.
+     */
+    private static Interview createEditedInterview(Interview interviewToEdit, LocalDateTime newDateTime) {
```
Comment (domlimm) [inline src/main/java/seedu/address/logic/commands/schedule/EditScheduleCommand.java:89]: I like this! Nice!

**Review by domlimm — APPROVED** [review]
Alright, doesn't seem to have any issues. Nice job leon! All LGTM. Good job on the extensive testing too! 😄

**Pull request merged**